File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        """
        since nums are sorted, can set two pointers on either side of array and inc/dec to get proper sum
        """
        i, j = 0, len(numbers) - 1
        while i != j:
            if numbers[i] + numbers[j] > target:
                j -= 1
            elif numbers[i] + numbers[j] < target:
                i += 1
            else:
                return [i + 1, j + 1]
        
                

File: 167-two-sum-ii-input-array-is-sorted/167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        return twoPointer(numbers, target);
    }
    
    public int[] twoPointer(int[] nums, int target) {
        int l = 0, r = nums.length - 1;
        
        while (l < r) {
            int sum = nums[l] + nums[r];
            if (sum < target) {
                l += 1;
            } else if (sum > target) {
                r -= 1;
            } else {
                return new int[] {l + 1, r + 1};
            }
        }
        
        return null;
    }
}